src/discordcr/rest.cr

```diff
@@ -81,7 +81,10 @@ module Discord
     def request(route_key : Symbol, major_parameter : UInt64?, method : String, path : String, headers : HTTP::Headers, body : String?)
       response = raw_request(route_key, major_parameter, method, path, headers, body)
 
-      raise StatusException.new(response) unless response.success?
+      unless response.success?
+        raise CodeException.new(response) if response.content_type == "application/json" && response.body.includes?(%("code")) && response.body.includes?(%("message"))
```

Comment:
I'm not particularly happy with the end bit, I would just check for `application/json`, try to instantiate `CodeException`, and rescue `JSON::ParseException` here.